### PR TITLE
Golang bindings sync support

### PR DESCRIPTION
**Pull request by haaawk**

### CODE REVIEW

File: crates/bindings/c/src/lib.rs [hunk 1]
```diff
@@ -3,11 +3,26 @@
...
 
+use std::{sync::{Once, Mutex}, borrow::BorrowMut};
+
+use tokio::runtime::Runtime;
 use types::{
     blob, libsql_connection, libsql_connection_t, libsql_database, libsql_database_t, libsql_row,
     libsql_row_t, libsql_rows, libsql_rows_future, libsql_rows_future_t, libsql_rows_t,
 };
 
+static mut RT: Option<Mutex<Runtime>> = None;
```
Comment (psarna) [inline crates/bindings/c/src/lib.rs:14]: Is Mutex needed here at all? It looks like you only ever call `block_on` on the runtime, and it takes `&self`, so doesn't need mutual exclusion.

File: crates/bindings/c/src/lib.rs [hunk 2]
```diff
@@ -3,11 +3,26 @@
...
+static INIT: Once = Once::new();
+
+fn get_rt<'a>() -> &'a Mutex<Runtime> {
```
Comment (psarna) [inline crates/bindings/c/src/lib.rs:17]: You might find lazy_static useful for exactly that purpose: https://docs.rs/lazy_static/latest/lazy_static/

File: crates/bindings/c/src/lib.rs (hunk 1 above)
Comment (haaawk) [reply crates/bindings/c/src/lib.rs:14]: I'm not sure if the Mutex is not helping in making sure those methods can be called from different threads. Will check.

Comment (haaawk) [reply crates/bindings/c/src/lib.rs:14]: It seems that Runtime is Sync anyway so I will drop the Mutex.

File: crates/bindings/c/src/lib.rs (hunk 2 above)
Comment (haaawk) [reply crates/bindings/c/src/lib.rs:17]: That worked. Thanks

File: crates/bindings/c/src/lib.rs [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 mod types;
 
+use tokio::runtime::Runtime;
 use types::{
     blob, libsql_connection, libsql_connection_t, libsql_database, libsql_database_t, libsql_row,
     libsql_row_t, libsql_rows, libsql_rows_future, libsql_rows_future_t, libsql_rows_t,
 };
 
+lazy_static! {
+    static ref RT: Runtime = tokio::runtime::Runtime::new().unwrap();
```
Comment (LucioFranco) [inline crates/bindings/c/src/lib.rs:15]: Do we want a full runtime? Or should we just create one with 1 worker thread?

Comment (haaawk) [reply crates/bindings/c/src/lib.rs:15]: I'm happy to do one with 1 worker thread if that's enough for the core Rust library.
Just tell me if 1 thread is enough for the replicator and how to create tokio::Runtime with 1 thread :).

This is a great comment but I will push it to follow-up as I'm going for vacation today and don't want a PR hanging for a week.

**Pull request merged**